Gem/Code/Source/Components/StressTestComponent.cpp

```diff
@@ -0,0 +1,140 @@
+ *
+ */
+
+#include <Source/Components/StressTestComponent.h>
+
+#include <Source/Components/NetworkAiComponent.h>
+#include <Source/Components/WasdPlayerMovementComponent.h>
+
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Multiplayer/IMultiplayer.h>
+#include <Multiplayer/Components/NetBindComponent.h>
+#include <Multiplayer/ConnectionData/IConnectionData.h>
+#include <Multiplayer/ReplicationWindows/IReplicationWindow.h>
+
+namespace MultiplayerSample
+{
+    void StressTestComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<StressTestComponent, StressTestComponentBase>()
+                ->Version(1);
+        }
+
+        StressTestComponentBase::Reflect(context);
+    }
+
+    void StressTestComponent::OnInit()
+    {
+    }
+
+    void StressTestComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+#ifdef IMGUI_ENABLED
+        ImGui::ImGuiUpdateListenerBus::Handler::BusConnect();
+#endif
+        auto agentType = AZ::Interface<Multiplayer::IMultiplayer>::Get()->GetAgentType();
+        switch (agentType)
+        {
+        case Multiplayer::MultiplayerAgentType::DedicatedServer:
+        case Multiplayer::MultiplayerAgentType::ClientServer:
+            m_isServer = true;
+            break;
+        default:
+            break;
+        }
+    }
+
+    void StressTestComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+#ifdef IMGUI_ENABLED
+        ImGui::ImGuiUpdateListenerBus::Handler::BusDisconnect();
+#endif
+    }
+
+#if defined(IMGUI_ENABLED)
+    void StressTestComponentController::OnImGuiMainMenuUpdate()
+    {
+        if (ImGui::BeginMenu("Multiplayer Sample"))
+        {
+            ImGui::Checkbox("Entity Spawner", &m_displayEntitySpawner);
+            ImGui::EndMenu();
+        }
+    }
+
+    void StressTestComponentController::OnImGuiUpdate()
+    {
+        if (m_displayEntitySpawner)
+        {
+            if (ImGui::Begin("Entity Spawner", &m_displayEntitySpawner, ImGuiWindowFlags_None))
+            {
+                DrawEntitySpawner();
+            }
+        }
+    }
+
+    void StressTestComponentController::DrawEntitySpawner()
+    {
+        ImGui::SliderInt("Quantity", &m_quantity, 1, 100);
+        ImGui::SliderInt("Team ID", &m_teamID, 0, 3);
+        if (ImGui::Button("Spawn AI Entity"))
+        {
+            for (int i = 0; i != m_quantity; ++i)
+            {
+                if (m_isServer)
+                {
+                    HandleSpawnAIEntity(nullptr, m_teamID);
+                }
+                else
+                {
+                    SpawnAIEntity(m_teamID);
+                }
+            }
+        }
+    }
+#endif
```

Comment:
If the `#if` and `#endif` are more than a few lines apart it's recommended to add the condition as a comment to the `#endif` like `#endif //defined(IMGUI_ENABLED)` as this makes it easier to read.